app/interactors/ratingWeets.js

```diff
@@ -0,0 +1,57 @@
+const { Rating, User, sequelize } = require('../models');
+const logger = require('../logger');
+const errors = require('../errors');
+const utilities = require('../helpers/utilities');
+
+const createRatingWeet = async (ragintgWeetData, weetUserId) => {
+  let transaction = {};
+  let rateWeetResult = {};
+  const { ratingUserId, weetId, score } = ragintgWeetData;
+  try {
+    transaction = await sequelize.transaction();
+
+    logger.info(
+      `ratingWeets-interactor::getting rating weet to update::ratingUserId::${ratingUserId}::weetId::${weetId}`
+    );
+    const ratingWeetToUpdate = await Rating.findOne({
+      where: { ratingUserId, weetId },
+      attributes: ['id', 'ratingUserId', 'weetId', 'score']
+    });
+
+    if (ratingWeetToUpdate) {
+      logger.info(
+        `ratingWeets-interactor::updating rating weet::ratingWeetToUpdate::${JSON.stringify(
+          ratingWeetToUpdate
+        )}`
+      );
+      ratingWeetToUpdate.score = score;
+      rateWeetResult = await ratingWeetToUpdate.save({ transaction });
+    } else {
+      logger.info(
+        `ratingWeets-interactor::creating rating weet::ragintgWeetData::${JSON.stringify(ragintgWeetData)}`
+      );
+      rateWeetResult = await Rating.create(ragintgWeetData, { transaction });
+    }
```

Comment:
I don't like this method longitude. It's too big and attack different objectives.
I think it could be split into some smaller methods.
Also, the query building is better to be in a service than here. Like having:
```js
service.findOneRating(ratingUserId, weetId);
```
And then the query definition en the service